src/pg_to_evalscript/javascript_common/common.js

```diff
@@ -3,81 +3,37 @@ function parse_rfc3339(dt, default_h = 0, default_m = 0, default_s = 0) {
     "^([0-9]{4})-([0-9]{2})-([0-9]{2})([Tt]([0-9]{2}):([0-9]{2}):([0-9]{2})(\\.[0-9]+)?)?(([Zz]|([+-])([0-9]{2}):([0-9]{2})))?";
   const regexDate = "^([0-9]{4})-([0-9]{2})-([0-9]{2})$";
 
-  let result = null;
-
   try {
     const g = dt.match(regexDateTime);
     if (g) {
-      let date = Date.UTC(
-        parseInt(g[1]), //year
-        parseInt(g[2]) - 1, // month
-        parseInt(g[3]), //day
-        parseInt(g[5] || default_h), //hour
-        parseInt(g[6] || default_m), //minute
-        parseInt(g[7] || default_s), // second
-        parseFloat(g[8]) * 1000 || 0 // milisecond
-      );
-
-      //for date-time strings either time zone or Z should be provided
+      // for date-time strings either time zone or Z should be provided
       if (g[5] !== undefined && g[9] === undefined) {
         return null;
       }
 
-      //check if timezone is provided
-      if (g[9] !== undefined && g[9] !== "Z") {
-        //offset in minutes
-        const offset =
-          (parseInt(g[12] || 0) * 60 + parseInt(g[13] || 0)) *
-          (g[11] === "+" ? -1 : 1);
-        //add offset in miliseconds
-        date = date + offset * 60 * 1000;
-      }
+      const date = new Date(dt);
 
       return {
         type: dt.match(regexDate) ? "date" : "date-time",
         value: new Date(date).toISOString(),
       };
     }
-  } catch (err) {
-    //
-  }
+  } catch (err) {}
 
-  return result;
+  return null;
 }
 
 function parse_rfc3339_time(t) {
   const regexTime = "(([0-9]{2}):([0-9]{2}):([0-9]{2})(\\.[0-9]+)?)?(([Zz]|([+-])([0-9]{2}):([0-9]{2})))";
   try {
     const g = t.match(regexTime);
     if (g) {
-      let date = Date.UTC(
-        0, // year
-        0, // month
-        1, // day
-        parseInt(g[2]), // hour
-        parseInt(g[3]), // minute
-        parseInt(g[4]), // second
-        parseFloat(g[5]) * 1000 || 0 // milisecond
-      );
-
-      // for time strings either time zone or Z should be provided
-      if (g[2] !== undefined && g[6] === undefined) {
-        return null;
-      }
-
-      // check if timezone is provided
-      if (g[6] !== undefined && g[6] !== "Z") {
-        // offset in minutes
-        const offset =
-          (parseInt(g[9] || 0) * 60 + parseInt(g[10] || 0)) *
-          (g[8] === "+" ? -1 : 1);
-        // add offset in miliseconds
-        date = date + offset * 60 * 1000;
-      }
+      const split = t.split(new RegExp('[Tt]'));
+      const time = split.length === 1 ? split[0] : split[1];
```

Comment:
Do we actually need to support parsing of just the time part of date-time? From the function name I would expect it to fail on strings that contain the date part.

Reply:
Parsing time from date-time strings is used for time based comparisons. With all datetimes being normalised to the same date it's simple to make time-only comparisons. 

Example: 
Input date-time:
```
"2022-03-16T03:30:00.000Z"
"2022-03-17T03:00:00.000Z"
"2022-03-20T04:30:00.000Z"
```

Interval to compare by:
```
["03:00:00Z", "04:00:00Z"]
```

Reply:
I see - I agree that such a function is useful; but I wonder whether it should really be called `parse_rfc3339_time`.. Vanilla implementation of what I intuitively expect when reading `parse_rfc3339_time` would assume that input is compatible with `full-time` or `partial-time` from [the RFC](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6); and would fail otherwise... The two use-cases we have for this[^1] are IMO distinct enough that we wouldn't want to mix them up in the same function.

Or we should at least make it clear from the name of the function that it doesn't really parse its input, but rather extracts information from a subset of its input.

[^1]: The two use-cases being a) I have a string that I assume is time-of-day and would like to convert it to a data object that I can work with vs. b) I have a string that I know is date-time and I would like to strip the date information to get a time-of-day data object

Reply:
I agree the naming is not nearly descriptive enough. But thinking about a better name I think also the purpose for such function is not intuitive at all.

What if instead of returning a date object normalised to `1900-01-01` we would rather return a number of seconds of the day. And name the function `extract_seconds_of_day()` or similar.

```
new Date('1970-01-01T' + matchTime[0]).getTime() / 1000
```

With this update I believe it would also make more sense to keep the two use cases together in the same function. As it would make sense to do so from the usage perspective.

Reply:
I agree (assuming that this is what we need), but maybe go with milliseconds, so that we can return an `uint` rather than floating-point number; and we retain precision of the input

Reply:
> assuming that this is what we need

I believe it should be ok. If we ever need to format extracted time, we can still construct new date object from milliseconds.